src/utils/destination-configuration.ts

```diff
@@ -0,0 +1,196 @@
+            oAuthTokenUrl: config.oAuthTokenUrl
+        });
+
+        const axiosRequestConfig = {
+            baseURL:
+                config.destinationUrl +
+                DestinationConfiguration.DESTINATION_PATH +
+                config.destinationName,
+            timeout: 2500,
+            retryConfig: {
+                maxRetries: 3,
+                retryBackoff: 1000
+            }
+        };
+
+        this.axiosInstance = axios.create(axiosRequestConfig);
+        setupAuthorizationHeaderOnRequestInterceptor(
+            this.axiosInstance,
+            Promise.resolve(oAuthAuthentication)
+        );
+    }
+
+    // eslint-disable-next-line require-jsdoc
+    async getAuthentication(): Promise<any> {
+        return new Promise((resolve, reject) => {
+            this.axiosInstance
+                .request({})
+                .then((destination) => {
+                    const destinationConfiguration = destination.data.destinationConfiguration;
+                    if (
+                        destinationConfiguration.Authentication ==
+                        DestinationConfiguration.BASIC_AUTHENTICATION
+                    ) {
+                        return resolve(
+                            new BasicAuthentication({
+                                username: destinationConfiguration.User,
+                                password: destinationConfiguration.Password
+                            })
+                        );
+                    }
+                    if (
+                        destinationConfiguration.Authentication ==
+                        DestinationConfiguration.OAUTH_2_CLIENT_CREDENTIALS
+                    ) {
+                        return resolve(
+                            new OAuthAuthentication({
+                                username: destinationConfiguration.clientId,
+                                password: destinationConfiguration.clientSecret,
+                                oAuthTokenUrl: destinationConfiguration.tokenServiceURL
+                            })
+                        );
+                    }
+                    if (
+                        destinationConfiguration.Authentication ==
+                        DestinationConfiguration.CLIENT_CERTIFICATE_AUTHENTICATION
+                    ) {
+                        return resolve(this.buildKeyStore(destination.data));
+                    }
+                })
+                .catch((error) => {
+                    return reject(error);
+                });
+        });
+    }
+
+    // eslint-disable-next-line require-jsdoc
+    private buildKeyStore(destination: any): KeyStore {
+        const keyStoreFormat = destination.certificates[0].Name.substr(
+            destination.certificates[0].Name.length - 3
+        );
+        if (keyStoreFormat == KeystoreFormat.JKS) {
+            return this.buildJksKeyStore(
+                destination.certificates[0].Content,
```

Comment:
Why do we get only the first certificate? Can there be more than one? Is this only the client certificate or the whole certificate chain?

Reply:
It is the whole certificate chain.